src/synapse/workload.py

```diff
@@ -870,28 +871,18 @@ def generate_nginx_config(container: ops.Container, main_unit_address: str) -> N
         container: Container of the charm.
         main_unit_address: Main unit address to be used in configuration.
     """
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/main_location.conf.template",
-            "/etc/nginx/main_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        ["sed", "-i", f"s/main-unit/{main_unit_address}/g", "/etc/nginx/main_location.conf"],
-    ).wait()
-    container.exec(
-        [
-            "cp",
-            "/etc/nginx/abuse_report_location.conf.template",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
-    container.exec(
-        [
-            "sed",
-            "-i",
-            f"s/main-unit/{main_unit_address}/g",
-            "/etc/nginx/abuse_report_location.conf",
-        ],
-    ).wait()
+    file_loader = FileSystemLoader(Path("./nginx_rock/etc"), followlinks=True)
```

Comment:
Since the template files are not used in the rock, wouldn't be better to move to somewhere else?

See the PostgreSQL charm as an example:
https://github.com/canonical/postgresql-k8s-operator/blob/167ce3ae7495b51c741af193863d58e9c1f8ef08/src/patroni.py#L340

Reply:
I actually did have something similar, decided to keep the templates where they were originally. i can move them under a `/templates` directory now